tests/unit/Core/Content/Mail/Service/MailServiceTest.php

```diff
@@ -254,4 +268,134 @@ public function testSendMailWithoutSenderName(): void
 
         static::assertInstanceOf(Email::class, $email);
     }
+
+    public function testMailSenderExceptionIsHandled(): void
+    {
+        $salesChannelId = Uuid::randomHex();
+
+        $salesChannel = new SalesChannelEntity();
+        $salesChannel->setId($salesChannelId);
+        $context = Context::createDefaultContext();
+
+        $salesChannelResult = new EntitySearchResult(
+            'sales_channel',
+            1,
+            new SalesChannelCollection([$salesChannel]),
+            null,
+            new Criteria(),
+            $context
+        );
+
+        $this->salesChannelRepository->expects($this->once())->method('search')->willReturn($salesChannelResult);
+
+        $data = [
+            'recipients' => [],
+            'senderName' => 'me',
+            'senderEmail' => 'me@shopware.com',
+            'subject' => 'Test email',
+            'contentPlain' => 'Content plain',
+            'contentHtml' => 'Content html',
+            'salesChannelId' => $salesChannelId,
+        ];
+
+        $email = (new Email())->subject($data['subject'])
+            ->html($data['contentHtml'])
+            ->text($data['contentPlain'])
+            ->to('me@shopware.com')
+            ->from(new Address($data['senderEmail']));
+
+        $this->mailFactory->expects($this->once())->method('create')->willReturn($email);
+        $this->templateRenderer->expects($this->exactly(4))->method('render')->willReturn('');
+
+        $this->logger->expects($this->once())->method('log')->with(Level::Error);
+
+        $beforeValidateEvent = null;
+        $mailErrorEvent = null;
+
+        $this->eventDispatcher
+            ->method('dispatch')
+            ->willReturnCallback(function (Event $event) use (&$beforeValidateEvent, &$mailErrorEvent) {
+                if ($event instanceof MailBeforeValidateEvent) {
+                    $beforeValidateEvent = $event;
+
+                    return $event;
+                }
+
+                $mailErrorEvent = $event;
+
+                return $event;
+            });
+
+        $this->mailSender->expects($this->once())->method('send')->willThrowException(new \Exception('Mail sending failed'));
+
+        $email = $this->mailService->send($data, Context::createDefaultContext());
+
+        static::assertNull($email);
+        static::assertNotNull($beforeValidateEvent);
+        static::assertInstanceOf(MailErrorEvent::class, $mailErrorEvent);
+        static::assertSame(Level::Error, $mailErrorEvent->getLogLevel());
+        static::assertNotNull($mailErrorEvent->getMessage());
+        static::assertSame('Could not send mail with error message: Mail sending failed', $mailErrorEvent->getMessage());
+        static::assertSame('Content html', $mailErrorEvent->getTemplate());
+        static::assertEmpty($mailErrorEvent->getTemplateData());
+    }
```

Comment:
Always triple check your backports 🤦‍♂️ . I was really confused why a test now suddenly fails but turns out the bot backported a test I didn't touched but didn't backported the implementation change (https://github.com/shopware/shopware/pull/8436).

I will remove the untouched test again...